File: java/mohit/Heap/Maximum Possible Product After K Operations.java
static int solve(int n, int[] nums, int k)
	{
		//CODE HERE 
		PriorityQueue<Integer> pq = new PriorityQueue<>();
		
		// O(nlogn)
		for(int i: nums)
		{
			pq.add(i);
		}
		
		for (int i = 0; i < k; i++) 
		{
			pq.add(pq.remove()+1);
		}
		
		long ans = 1;
		
		while(!pq.isEmpty())
		{
			ans = (ans*pq.remove()) % 1000000007;
		}
		
		return (int)ans;
	}

File: Python/chandu/Maximum_Possible_Product_After_K_Operations.py
import heapq
def solve(n, nums, k):
    # CODE HERE
    heapq.heapify(nums)
    for i in range(k):
        num = heapq.heappop(nums)+ 1
        heapq.heappush(nums, num)

    ans = 1
    for i in nums:
        ans *= i
    
    return ans%1000000007
